pkg/types/ordermap.go

```diff
@@ -125,28 +125,36 @@ func (m *SyncOrderMap) Remove(orderID uint64) (exists bool) {
 	return exists
 }
 
-func (m *SyncOrderMap) Add(o Order) {
+func (m *SyncOrderMap) processPendingRemoval() {
+	if len(m.pendingRemoval) == 0 {
```

Comment:
`pendingRemoval` should be protected by the lock as well?

Reply:
it’s read only, should be okay?